conversion.go

```diff
@@ -16,8 +16,8 @@ import (
 //   - If the conversion is not possible for the desired type, an [ErrUnsupportedConversion] error is wrapped in the returned error.
 //   - If the conversion fails from string, an [ErrStringConversion] error is wrapped in the returned error.
 //   - If the conversion results in an error, an [ErrConversionIssue] error is wrapped in the returned error.
-func Convert[NumOut Number](orig any) (converted NumOut, err error) {
-	switch v := orig.(type) {
+func Convert[NumOut Number, NumIn Input](orig NumIn) (converted NumOut, err error) {
+	switch v := any(orig).(type) {
```

Comment:
What if `orig` is of a type `A` declared as `type A int`? In that case, `A` satisfies `Input` so it compiles, but I believe this switch statement won't find a match and hence this function returns an error.

Reply:
Yes, it's the first step of a refactoring.

Here I'm only applying a part of the refactoring that need to be done.

I'm only addressing the "the function accepts any" issues

I still have to deal with "type alias are not working" issue.

I have code ready locally.

But I wanted this PR first.